sbc_translation/models/correspondence.py

```diff
@@ -194,6 +152,88 @@ def update_translation(self, translate_lang, translate_text, translator):
             self.letter_image.unlink()
             super(Correspondence, self).process_letter()
 
+    ##########################################################################
+    #                             PRIVATE METHODS                            #
+    ##########################################################################
+    def _get_translation_langs(self):
+        """
+        Finds the source_language et destination_language suited for
+        translation of the given letter.
+
+        S2B:
+            - src_lang is the original language of the letter
+            - dst_lang is the lang of the child if translatable, else
+              english
+
+        B2S:
+            - src_lang is the original language if translatable, else the
+              current translated language of the letter (mostly english)
+            - dst_lang is the main language of the sponsor
+        :return: src_lang, dst_lang
+        :rtype: res.lang.compassion, res.lang.compassion
+        """
+        self.ensure_one()
+        src_lang_id = False
+        dst_lang_id = False
+        if self.direction == 'Supporter To Beneficiary':
+            # Check that the letter is not yet sent to GMC
+            if self.kit_identifier:
+                raise Warning(_("Letter already sent to GMC cannot be "
+                                "translated! [%s]") % self.kit_identifier)
+
+            src_lang_id = self.original_language_id
+            child_langs = self.beneficiary_language_ids.filtered(
+                'translatable')
+            if not child_langs:
+                raise Warning(_("Child %s has no translatable language")
+                              % self.child_id.code)
+            dst_lang_id = child_langs[-1]
```

Comment:
Normally English is always in the child languages, however you are right, it is preferable to set it so that in case it is not set it won't fail. I will change this.